docs/source/conceptual/guidance.md

```diff
@@ -1,425 +1,64 @@
 
-If you're not up to date, grab the latest version and let's get started!
+- a specific JSON object
+- a function signature
+- typed output like a list of integers
 
-## How it works
+However these use cases can span a wide range of applications, such as:
 
-TGI leverages the [outlines](https://github.com/outlines-dev/outlines) library to efficiently parse and compile the grammatical structures and tools specified by users. This integration transforms the defined grammars into an intermediate representation that acts as a framework to guide and constrain content generation, ensuring that outputs adhere to the specified grammatical rules.
+- extracting structured data from unstructured text
+- summarizing text into a specific format
+- limit output to specific classes of words (act as a LLM powered classifier)
+- generate the input to specific APIs or services
+- provide reliable and consistent output for downstream tasks
+- extract data from multimodal inputs
 
-## Table of Contents 📚
 
-### Grammar and Constraints
+## How it works?
 
-- [The Grammar Parameter](#the-grammar-parameter): Shape your AI's responses with precision.
-- [Constrain with Pydantic](#constrain-with-pydantic): Define a grammar using Pydantic models.
-- [JSON Schema Integration](#json-schema-integration): Fine-grained control over your requests via JSON schema.
-- [Using the client](#using-the-client): Use TGI's client libraries to shape the AI's responses.
+Diving into the details, guidance is enabled by including a grammar with a generation request that is compiled, and used to modify the chosen tokens.
 
-### Tools and Functions
+This process can be broken down into the following steps:
 
-- [The Tools Parameter](#the-tools-parameter): Enhance the AI's capabilities with predefined functions.
-- [Via the client](#text-generation-inference-client): Use TGI's client libraries to interact with the Messages API and Tool functions.
-- [OpenAI integration](#openai-integration): Use OpenAI's client libraries to interact with TGI's Messages API and Tool functions.
+1. A request is sent to the backend, it is processed and placed in batch. Processing includes compiling the grammar into a finite state machine and a grammar state.
+2. The model does a forward pass over the batch. This returns probabilities for each token in the vocabulary for each request in the batch.
 
-## Grammar and Constraints 🛣️
+3. The process of choosing one of those tokens is called `sampling`. The model samples from the distribution of probabilities to choose the next token. In TGI all of the steps before sampling are called `processor`. Grammars are applied as a processor that masks out tokens that are not allowed by the grammar.
 
-### The Grammar Parameter
+<div class="flex justify-center">
+    <img
+        class="block dark:hidden"
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/logit-grammar-mask.gif"
+    />
+    <img
+        class="hidden dark:block"
```

Comment:
These gifs are great! I also liked this [high-level graph](https://huggingface.slack.com/archives/C05CFK1HM0T/p1714157053184219?thread_ts=1714052150.677459&cid=C05CFK1HM0T) on slack

Reply:
@aymeric-roucher **updated and remade the other steps of the diagram as gifs. Please let me know if its missing anything needed from the original